musicTheory.ts

```diff
@@ -300,6 +308,28 @@ const getAccidentals = (num: number): Array<Pitch> => {
   }
 };
 
+/*  REVIEW
+    What if this function is called with the wrong number for param numSharpFlats?
+
+    There's nothing enforcing the correct number so for example I can call 
+    const keyOfC = getKey(1, { note: "C", accidental: null }, "Ionian");
+    and it'll give me a C scale with an F#.
+
+    Though we implicitly know the number of sharps/flats needed for a scale from music knowledge, 
+    it's still a point for a potential bug to appear if for example, another dev tries using the 
+    function but doesn't have the same knowledge, thus doesn't know what number use for the param.
+
+    Not having the param for the number of sharps and flats complicates the problem a lot, but if
+    you want to keep this function as is, then perhaps introducing a map to ensure correctness 
+    and updating the type for numsharpFlats would be good. e.g.
+    const SharpFlatMap: Record<Pitch['note'], number> = 
+    {
+      C: 0
+      G: 1
+      D: 2
+      ...etc
+    }
+*/
```

Comment:
Yeah, I was torn between how we will identify keys. I threw in the number of sharps and flats, and the mode, but really most of the time we will want something like A major, or B minor rather than needing to say how many sharps or flat we want. Personally, I think of the number of sharps and flats when I think of key, and the starting pitch as the mode. So 3 sharps starting on “1” is A major. 3 sharps starting on the 6th scale degree is F# minor. 

But since I am exporting this function, you are totally right that it’s easy to misuse. If the PDFs we read in are going to have a “key symbol” (something like “Bb”), we are probabyl going to want to make this function private, and provide a function that can’t be used wrong for the publuc facing interface. Ideally also a function that doesn’t mind seeing “Bb” and knowing we mean 2 falts ionian. 

What do you think of making getKey private, and adding a “wrapper function” that expects chord symbols more like the ones we expect to see in the chord sheets we are going to transpose, and passes the correct arguments to this getKey?